hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/cdc/HoodieCDCRDD.scala

```diff
@@ -561,7 +561,7 @@ class HoodieCDCRDD(
       originTableSchema.structTypeSchema.zipWithIndex.foreach {
         case (field, idx) =>
           if (field.dataType.isInstanceOf[StringType]) {
-            map(field.name) = record.getString(idx)
+            map(field.name) = Option(record.getUTF8String(idx)).map(_.toString).orNull
           } else {
```

Comment:
Can we add a simple UT for it.

Reply:
Absolutely, the only problem is that I don't see any unit tests for the cdc package so it's hard to follow existing examples. I tried implementing a test that extends `HoodieClientTestBase` but that was getting me far from the requested "unit test". What would be the best way to start with tests for this particular issue?

Reply:
This is what I have based on my limited knowledge of Hudi: https://github.com/apache/hudi/pull/9064/commits/c88aee0f26afa779594a9981d86aeb3d06727d4b

I'm more than happy to make further adjustments when needed.

Reply:
I think we are cool, a basic tool test makes sense to me. It's cool if we can make the tool a singleton, no json mapper resigtering for each invocation.

Reply:
Is [this](https://github.com/apache/hudi/pull/9064/commits/af87c98dd4c370bb40287013adcecd314e20b546) better or would you like me go further?

Reply:
Looks good to me ~